aodntools/timeseries_products/hourly-timeseries.py

```diff
@@ -0,0 +1,554 @@
+from __future__ import print_function
+import sys
+import os.path
+from dateutil.parser import parse
+from datetime import datetime
+import json
+from netCDF4 import Dataset
+import argparse
+
+import numpy as np
+import xarray as xr
+import pandas as pd
+
+
+def sort_files_to_aggregate(files_to_aggregate):
+    """
+    sort the list of files to aggregate by time_deployment start attribute
+
+    :param files_to_aggregate: list of file URLs
+    :return: list of file URLs
+    """
+    file_list_dataframe = pd.DataFrame(columns=["url", "deployment_date"])
+    for file in files_to_aggregate:
+        with Dataset(file) as nc:
+            try:
+                file_list_dataframe = file_list_dataframe.append({'url': file,
+                                                                  'deployment_date': parse(
+                                                                      nc.getncattr('time_deployment_start'))},
+                                                                 ignore_index=True)
+            except:
```

Comment:
I've had some files with no `time_deployment_start` attribute, This is what I'm catching.
exception class added.

Reply:
Ok, thanks. Rather than raise an error here, we should check for this in `check_file()`. That way we can and add the file to the rejected_files list if the attribute is missing, and get on with the rest of the product!

While we're at it, let's check for `time_deployment_end` too, since we also need it to trim the data to in-water only!